services/algolia-sync/src/components/graphql/query.js

```diff
@@ -0,0 +1,22 @@
+const createClient = require('./create-client');
+require('dotenv').config();
+
+const queryFromBase = async (query, tenantKey, input) => {
+  const context = {
+    headers: {
+      'x-tenant-key': tenantKey,
+      'x-cdn-image-hostname': 'base.imgix.net',
+    },
+  };
+
+  try {
+    const apollo = createClient(process.env.GRAPHQL_URI);
```

Comment:
You are creating a new apollo client each time this function is executed. A better approach would be to define your client outside the scope of the method you're exporting so you can reuse the previously initialized client. Will save resources and speed up execution!

Reply:
Since this is executing on the server you do indeed want to create a new Apollo client on each request, otherwise query cache will live past the scope of request.